port/osx/omrosdump.c

```diff
@@ -425,33 +427,34 @@ omrdump_create(struct OMRPortLibrary *portLibrary, char *filename, char *dumpTyp
 	kr = task_get_bootstrap_port(mach_task_self(), &special_port);
 	if (KERN_SUCCESS != kr) {
 		mach_error("failed get special port:\n", kr);
-		return kr;
+		goto done;
 	}
 	pass_port = mach_task_self();
 	/* pass parent task port to child through special port inheritance */
 	kr = task_set_bootstrap_port(mach_task_self(), pass_port);
 	if (KERN_SUCCESS != kr) {
 		mach_error("failed set special port:\n", kr);
-		return kr;
+		goto done;
 	}
 
 	child_pid = fork();
 	if (0 == child_pid) { /* in child process */
 		kr = task_get_bootstrap_port(mach_task_self(), &pass_port);
 		if (KERN_SUCCESS != kr) {
 			mach_error("failed get special port:\n", kr);
-			return kr;
+			goto done_child_proc;
 		}
 		/* Move to specified folder before dumping */
 		if (NULL != lastSep) {
 			/* keep separator for cases such as when the path is '/' */
 			lastSep[1] = '\0';
 			if (0 != chdir(filename)) {
 				perror("failed to change directories for dump");
-				return KERN_FAILURE;
+				goto done_child_proc;
```

Comment:
I think it's better to just ignore this failure and attempt to write the dump in the current directory.

Reply:
Changed it as you suggested, if `chdir` fails it will attempt to create a core dump in the current working directory.